packages/core/src/video/VideoForRendering.tsx

```diff
@@ -65,24 +65,26 @@ const VideoForRenderingForwardFunction: React.ForwardRefRenderFunction<
 	});
 
 	useEffect(() => {
-		if (!props.src) {
+		if (!props.src && !props.children) {
 			throw new Error('No src passed');
 		}
 
 		if (props.muted) {
 			return;
 		}
 
-		registerAsset({
-			type: 'video',
-			src: getAbsoluteSrc(props.src),
-			id,
-			frame: absoluteFrame,
-			volume,
-			isRemote: isRemoteAsset(getAbsoluteSrc(props.src)),
-			mediaFrame: frame,
-			playbackRate: playbackRate ?? 1,
-		});
+		if (props.src) {
+			registerAsset({
+				type: 'video',
```

Comment:
We need to fix it, if you pass a <source> now, then the audio of the video will not play in the final video.

Do we know of a reliable way to determine the source of the video if children have been passed?

Reply:
as `<video>` may have multiple `<source>`, should we take the `src` of the first child?

Reply:
@mikaoelitiana, when I was exploring this feature some weeks ago, I had the feeling that we can leverage the DOM events on the `audio` and `video` tags to find which source was chosen.
Then, we can pass that source to Remotion internals (as we currently do with `props.src` only).

The [`loadedmetadata`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/loadedmetadata_event) event fires when the browser has chosen a source and [`mediaElement.currentSrc`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/currentSrc) will contain the chosen source.